mne/io/pick.py

```diff
@@ -374,6 +372,15 @@ def pick_types(info, meg=True, eeg=False, stim=False, eog=False, ecg=False,
     """
     # NOTE: Changes to this function's signature should also be changed in
     # PickChannelsMixin
+    if meg is None:
+        meg = True  # previous default arg
+        meg_default_arg = True  # default argument for meg was used
+    else:
+        meg_default_arg = False
+    # only issue deprecation warning if there are MEG channels in the data and
```

Comment:
Indeed this looks like it should work, but I would write a little 10 line test with read_info to make sure / assert it is the case (can use pytest.deprecated_call for ensuring it warns at the right time)

Reply:
Done and works, except for "planar1" and "planar2" which I need to handle correctly. However, I don't understand what these values do. In the test, I have `info1 = create_info(6, 256, ["eeg", "mag", "grad", "misc", "grad", "hbo"])`. Why do I get `[2]` for `mne.pick_types(info1, meg="planar1")` and `[]` for `mne.pick_types(info1, meg="planar2")`?

Reply:
> Why do I get [2] for mne.pick_types(info1, meg="planar1") and [] for mne.pick_types(info1, meg="planar2")?

There are two types of grads in Neuromag systems (one computes the gradient in the X direction, the other Y). When you pass `meg='grad'` in `create_info`, it creates a `planar1` variant, not `planar2`. I wouldn't bother adding a `planar2`, it should be good enough to assert that you you do get `[2]` for `planar1` and `[]` for `planar2` in your example.

Reply:
But wait, what about index 4? That's also a 'grad' channel - why is it neither `'planar1'` nor `'planar2'`?

Reply:
Ah that's weird. Indeed it should be one or the other. Must be a bug somewhere in that function I guess. Do you get `[2, 4]` when you do `meg='grad`, and/or `[1, 2, 4]` when you do `meg=True`?

Reply:
Yes, that's exactly what I get for these two queries.

Reply:
Want me to look into it, or do you want to fix it here?

Reply:
Please go for it!

Reply:
Okay, will look tomorrow if you don't beat me to it